docs/user_guide/building_blocks/config_objects.md

```diff
@@ -1,31 +1,103 @@
+Config must correspond to the
+processes defined for the datasets and added to the Campaign object associated to the Config.
+It is possible to get all root processes from a specific campaign using the
+{py:func}`~columnflow.config_util.get_root_processes_from_campaign()` function from columnflow.
+Examples of information carried by a process could be the cross section of the process, registered
+under the {external+order:py:attr}`order.process.Process.xsecs` attribute and further used for normalization_weights in columnflow, and a color for the plotting scripts, which can be set using
+the {external+order:py:attr}`order.mixins.ColorMixin.color1` attribute of the process.
+An example of a Process definition is given in the {ref}`Analysis, Campaign and Config <analysis_campaign_config>` section of the columnflow documentation.
+More informations about processes can be found in the
+{external+order:py:class}`order.process.Process` and the
+{external+order:doc}`quickstart` sections of the order documentation.
 
-TODO
 
-## "Variable" creation
+### Datasets
+
+The actual datasets to be processed in the analysis. These should be saved as objects of the
+{external+order:py:class}`order.dataset.Dataset` class and added to the Config object using its
+{external+order:py:meth}`order.config.Config.add_dataset()` method. The datasets added to the
+Config object must correspond to the
+datasets added to the Campaign object associated to the Config object. They are accessible through
+the {external+order:py:meth}`order.config.Campaign.get_dataset()` method of the Campaign class. The
+Dataset objects should contain for example informations on the number of files and number of events
+present in a Dataset as well as its key and if it is data or a Monte-Carlo dataset. It is also
+possible to change informations of a dataset in the config script. An example would be reducing the
+number of files to process for test purposes in a specific test config. This could be done with the
+following lines of code:
+e.g.
+```python
+n_files_max = 5
+for info in dataset.info.values():
+    info.n_files = min(info.n_files, n_files_max)
+```
+
+Once the processes and datasets have both been added to the config, one can check that the root
+process of all datasets is part of any of the registered processes, using the columnflow function
+{py:func}`~columnflow.config_util.verify_config_processes()`.
+
+An example of a Dataset definition is given in the {ref}`Analysis, Campaign and Config <analysis_campaign_config>` section of the columnflow documentation.
+
+
+### Variables
 
 In order to create histograms out of the processed datasets, columnflow uses
 {external+order:py:class}`order.variable.Variable`s. These
 Variables need to be added to the config using the
 function {external+order:py:meth}`order.config.Config.add_variable`. The standard syntax is as
-follows:
+follows for `cfg` the Config object:
 
 ```python
-config.add_variable(
+cfg.add_variable(
```

Comment:
done for the change, not quite sure how to add it from the example config, as the lines go like this: 

```
cfg.add_variable(
    name="jets_pt",
    expression="Jet.pt",
    binning=(40, 0.0, 400.0),
    unit="GeV",
    x_title=r"$p_{T} of all jets$",
)
cfg.add_variable(
    name="jet1_pt",
    expression="Jet.pt[:,0]",
    null_value=EMPTY_FLOAT,
    binning=(40, 0.0, 400.0),
    unit="GeV",
    x_title=r"Jet 1 $p_{T}$",
)
```

Do you know a way to select only the variable I am interested in? I could give the lines but this might change for every change in the config, so I would rather not do that...

Reply:
As far as I am aware, this works only on the first and last line of what you want to include, but this would be 
`cfg.add_variable(` and `)`, which is repeated for every variable and would then probably only take the first one, while I would prefer to have another one as example, e.g. with a null value.

Reply:
added comments in template, now done

Reply:
Thank you for this update! I'm sorry I'm nagging so much, but:
- I wouldn't write the pseudo code you've included with the same argumentation as before - this is not testable and might be outdated at some point. Alternatively, you could e.g. add comments in the template that describe the different keyword arguments in more detail.
- I think there is a `)` missing in the include. Can you fix this? I.e. either try the `end-after` keyword or add an additional comment to use as an achor point